clang/test/Analysis/stream-errno.c

```diff
@@ -248,3 +248,25 @@ void check_fflush_all(void) {
     if (errno) {}                    // no-warning
   }
 }
+
+void check_opendir(const char *Path) {
+  DIR *Dir = opendir(Path);
+  if (Dir == NULL) {
+    clang_analyzer_eval(errno != 0); // expected-warning{{TRUE}}
+    if (errno) {}                    // no-warning
+  } else {
+    if (errno) {}                    // expected-warning{{An undefined value may be read from 'errno'}}
+    closedir(Dir);
+  }
+}
+
+void check_fdopendir(int Fd) {
+  DIR *Dir = fdopendir(Fd);
+  if (Dir == NULL) {
+    clang_analyzer_eval(errno != 0); // expected-warning{{TRUE}}
+    if (errno) {}                    // no-warning
+  } else {
+    if (errno) {}                    // expected-warning{{An undefined value may be read from 'errno'}}
+    closedir(Dir);
+  }
+}
```

Comment:
I think for now it is not necessary to add tests for these functions. These functions are the same type as all other functions in `StdLibraryFunctionsChecker` that have currently no test. And it is questionable if these functions belong to stream handling functions or to `StreamChecker`.

Reply:
I have removed those tests. Do you mean all type `DIR` related functions are controversial now, so they need tests? Or there are some more functions are so?

Reply:
They need no tests like there is no test for `chdir`, `chown`, `close`, `open`, and all other functions in the checker. These `DIR` type functions are not more special than the others. It can be possible to add these `errno` tests for all of the functions in a separate file and in a generated way. The tests in **stream_*** files are related to `StreamChecker`, but not the `DIR` functions. (Goal of `StreamChecker` is now only to work with the functions with `FILE*` argument.)